aten/src/ATen/core/IdWrapper.h

```diff
@@ -22,7 +22,7 @@ namespace at {
  * for you, given the underlying type supports it.
  */
 template <class ConcreteType, class UnderlyingType>
-class AT_CORE_API IdWrapper {
```

Comment:
I have a feeling you're going to want to leave this, but maybe use AT_CORE_EXPORT since it's templated.

Reply:
@orionr I agree. This should be removed.

Also, I noticed there're CAFFE2_API or CAFFE2_EXPORT in several places with template.
I listed part of them in #11114 

Is there any reason to use these macros for defined-in-header classes/functions?
My understanding is, if something is defined in header, each compilation unit gets their own copy directly so it'll never be on library boundary.
In fact, we're suppose to hide their visibility to avoid redefinition, rather than expose them.

Reply:
@xkszltl it's a good point. Templates plus DLL boundaries are complex. There are definitely cases where we should explicitely do CAFFE2_EXPORT in the cc/cpp file because we are providing instantiations of the template - I seem to recall some of this is in ArgumentHelper - but if the implementation is exclusively in the header you are right. I'll take a look at the PR you reference.